sys/kern/src/arch/riscv64.rs

```diff
@@ -281,6 +273,7 @@ pub unsafe extern "C" fn _start_trap() {
         sd a1,  31*8(a0)    # store mepc for resume
         csrr a1, mscratch
         sd a1, 9*8(a0)      # store a0 itself
+        csrw mscratch, a0   # restore task ptr in mscratch
```

Comment:
Not sure how much it all matters, but presumably you could have made line 274 csrrw a1, mscratch, a0.

Reply:
Good point. I can go ahead and push that as a follow-up change in the PR.

Reply:
> Is there a humility change needed for this? CURRENT_TASK_PTR can no longer be used. 

Yes, there is a change required in humility. Drew raised a [bug](https://http://go/ji/SW-600) to track that.

> Similarly, do we now need to indicate somehow in _start_trap() that current task cannot be relied upon in mscratch between lines 240 and 276?

Are you thinking of conditions like exception happening somewhere in between lines 240-276?

Reply:
Pushed follow-up commit to address review comments.

Reply:
I was just thinking any sort of breakpoint or whatever when we are in this sequence and humility wants to interrogate the current running task.

Reply:
Ah I see. That is interesting. I will make a note of this on the bug as well so that we can think about this when we start adding support for humility.

Reply:
Done.